### PR TITLE
V1 queue vs V1

### PR DESCRIPTION
**Pull request by eladmallel**

### CODE REVIEW

File: src/Payer.sol [hunk 1]
```diff
@@ -21,83 +21,147 @@ import { Ownable } from 'openzeppelin-contracts/contracts/access/Ownable.sol';
...
+
+            uint256 _debtAmount = debt.amount;
+            address _debtAccount = debt.account;
+
+            if (amount < _debtAmount) {
+                // Not enough to cover entire debt, pay what you can and leave
+                debt.amount -= amount; // update remaining debt
+                totalDebt -= amount;
+                paymentToken.safeTransfer(_debtAccount, amount);
+                emit PaidBackDebt(_debtAccount, amount, false);
```
Comment (eladmallel) [inline src/Payer.sol]: could be nicer UX for event users to emit the remaining debt instead of a boolean, wdyt?

File: test/DebtQueue.t.sol [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import 'forge-std/Test.sol';
+import { DebtQueue } from '../src/libs/DebtQueue.sol';
+
+contract DebtQueueTest is Test {
+    using DebtQueue for DebtQueue.DebtDeque;
+
+    DebtQueue.DebtDeque public queue;
+
+    function test_popFront() public {
```
Comment (eladmallel) [inline test/DebtQueue.t.sol:12]: if we're already adding sanity tests here, perhaps worth adding a test with multiple items in the queue, showing the pop indeed pops the first item added?

File: test/TokenBuyer.t.sol [hunk 3]
```diff
@@ -201,8 +199,28 @@ contract TokenBuyerTest is Test, IBuyETHCallback {
...
     }
 
+    function test_buyETH_paysBackDebt() public {
+        // user has debt of 2000 tokens
+        vm.prank(owner);
+        payer.sendOrRegisterDebt(user, 2000e18);
+        assertEq(payer.debtOf(user), 2000e18);
+
+        // bot buys ETH for 2000 tokens
+        priceFeed.setPrice(2000 * 1e18);
```
Comment (eladmallel) [inline test/TokenBuyer.t.sol]: sometimes we have `2000 * 1e18` and sometimes we have `2000e18`. do you want to keep these different styles or stick with one of them?

File: script/DeployUSDC.s.sol [hunk 4]
```diff
@@ -20,12 +19,12 @@ contract DeployUSDCScript is Script {
     uint8 constant MAINNET_USDC_DECIMALS = 6;
 
     // PriceFeed config
-    address constant MAINNET_USDC_ETH_CHAINLINK = 0x986b5E1e1755e3C2440e960477f25201B0a8bbD4;
-    uint256 constant USDC_ETH_CHAINLINK_HEARTBEAT = 24 hours;
-    uint256 constant PRICE_UPPER_BOUND = 0.01e18; // i.e. 100 tokens buy 1 ETH
-    uint256 constant PRICE_LOWER_BOUND = 0.00001e18; // i.e. 100K tokens buy 1 ETH
+    address constant MAINNET_ETH_USDC_CHAINLINK = 0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419;
```
Comment (eladmallel) [inline script/DeployUSDC.s.sol]: maybe good to rename to ETH_USD (drop the C)
same for the heartbeat const name

File: script/DeployUSDC.s.sol [hunk 5]
```diff
@@ -85,23 +76,57 @@ contract DeployUSDCRinkeby is DeployUSDCScript {
...
 
-        iou.grantRole(iou.MINTER_ROLE(), address(payer));
-        iou.grantRole(iou.BURNER_ROLE(), address(payer));
+        new TokenBuyer(
+            usdc,
+            priceFeed,
+            USD_POSITION_IN_USD * 10**DECIMALS, // baselinePaymentTokenAmount
+            0, // minAdminBaselinePaymentTokenAmount
+            2 * USD_POSITION_IN_USD * 10**DECIMALS, // maxAdminBaselinePaymentTokenAmount
+            0, // botIncentiveBPs
```
Comment (eladmallel) [inline script/DeployUSDC.s.sol]: nit: reminder to rename these inline comments to `botDiscountBPs` etc.

File: src/Payer.sol (hunk 1 above)
Comment (davidbrai) [reply src/Payer.sol]: good idea. done 32f6ce855ce99b4254e295132ed975f327daa0f4

File: test/DebtQueue.t.sol (hunk 2 above)
Comment (davidbrai) [reply test/DebtQueue.t.sol:12]: added f44c74940f9acce59e5c81ab00f92d9d76a4373c

File: test/TokenBuyer.t.sol (hunk 3 above)
Comment (davidbrai) [reply test/TokenBuyer.t.sol]: normalized 17442dfbc75adc7d26978b68618ba0b508f8cf01

File: script/DeployUSDC.s.sol (hunk 4 above)
Comment (davidbrai) [reply script/DeployUSDC.s.sol]: fixed

File: script/DeployUSDC.s.sol (hunk 5 above)
Comment (davidbrai) [reply script/DeployUSDC.s.sol]: fixed

**Pull request closed without merging**